### ISSUE TITLE
Non-inline modules not found in crate hierarchy

### ISSUE THREAD

**Issue by Lunderberg** [rust-lang/rust-analyzer#15965]
Module files are erroneously marked as not within the crate hierarchy, with the majority of LSP functionality disabled, when the module declaration occurs within the body of a function.

```rust
// main.rs
fn main() {
    #[path = "my_mod.rs"]
    mod my_mod;
    my_mod::func();
}

// my_mod.rs
pub fn func() {}
```

Expected behavior: LSP works within both `main.rs` and `my_mod.rs`.

Actual behavior: The first two characters in `my_mod.rs` have a warning: `"file not included in crate hierarchy"`, and most LSP functionality, including finding of references, is non-functional within `my_mod.rs`.  LSP does work within `main.rs`, including jump-to-definition on `my_mod::func()`.

**rust-analyzer version**: Tested on both `rust-analyzer 1.74.0 (79e9716 2023-11-13)` and `rust-analyzer 1.76.0-nightly (37b2813 2023-11-24)`.

**rustc version**: Tested on both `rustc 1.74.0 (79e9716c9 2023-11-13)` and `rustc 1.76.0-nightly (37b2813a7 2023-11-24)`.

**relevant settings**: N/A

**Comment by Veykril** [comment]
Oh god, people actually use local modules?

I don't think we can reasonably support this. To be able to know this information we'd need to eagerly calculate all block def maps which is exactly what we don't wanna do ...
